Cubical/Algebra/Group/Abelianization/AbelianizationAsCoeq.agda

```diff
@@ -0,0 +1,354 @@
+                       incAb ((inv c · inv b) · inv a)        ≡⟨ cong incAb (sym (lid ((inv c · inv b) · inv a))) ⟩
+                       incAb (1g · ((inv c · inv b) · inv a)) ≡⟨ comm 1g (inv c · inv b) (inv a) ⟩
+                       incAb (1g · (inv a · (inv c · inv b))) ≡⟨ cong incAb (lid (inv a · (inv c · inv b))) ⟩
+                       incAb (inv a · (inv c · inv b))        ≡⟨ comm (inv a) (inv c) (inv b) ⟩
+                       incAb (inv a · (inv b · inv c))        ≡⟨ cong incAb (sym (lid (inv a · (inv b · inv c)))) ⟩
+                       incAb (1g · (inv a · (inv b · inv c))) ≡⟨ comm 1g (inv a) (inv b · inv c) ⟩
+                       incAb (1g · ((inv b · inv c) · inv a)) ≡⟨ cong incAb (lid ((inv b · inv c) · inv a)) ⟩
+                       incAb ((inv b · inv c) · inv a)        ≡⟨ cong (λ x → incAb (x · inv a)) (sym (invDistr c b)) ⟩
+                       incAb (inv (c · b) · inv a)            ≡⟨ cong incAb (sym (invDistr a (c · b))) ⟩
+                       incAb (inv (a · (c · b))) ∎)
+
+  assocAb : (x y z : Abelianization) → x ·Ab (y ·Ab z) ≡ (x ·Ab y) ·Ab z
+  assocAb =
+    elimProp3
+      (λ x y z → squash (x ·Ab (y ·Ab z))
+      ((x ·Ab y) ·Ab z))
+      (λ x y z → cong incAb (assoc x y z))
+
+  ridAb : (x : Abelianization) → x ·Ab 1Ab ≡ x
+  ridAb =
+    elimProp
+      (λ x → squash (x ·Ab 1Ab) x)
+      (λ x → cong incAb (rid x))
+
+  rinvAb : (x : Abelianization) → x ·Ab (invAb x) ≡ 1Ab
+  rinvAb =
+    elimProp
+      (λ x → squash (x ·Ab (invAb x)) 1Ab)
+      (λ x → (incAb x) ·Ab (invAb (incAb x)) ≡⟨ refl ⟩
+             (incAb x) ·Ab (incAb (inv x))   ≡⟨ refl ⟩
+             incAb (x · (inv x))             ≡⟨ cong incAb (fst (inverse x)) ⟩
+             incAb 1g                        ≡⟨ refl ⟩
+             1Ab ∎)
+
+  commAb : (x y : Abelianization) → x ·Ab y ≡ y ·Ab x
+  commAb =
+    elimProp2
+      (λ x y → squash (x ·Ab y) (y ·Ab x))
+      (λ x y → (incAb x) ·Ab (incAb y) ≡⟨ refl ⟩
+               incAb (x · y)           ≡⟨ cong incAb (sym (lid (x · y))) ⟩
+               incAb (1g · (x · y))    ≡⟨ comm 1g x y ⟩
+               incAb (1g · (y · x))    ≡⟨ cong incAb (lid (y · x)) ⟩
+               incAb (y · x)           ≡⟨ refl ⟩
+               (incAb y) ·Ab (incAb x) ∎)
+
+  -- The proof that the abelianization is in fact an abelian group.
+  asAbelianGroup : AbGroup ℓ
+  asAbelianGroup = makeAbGroup 1Ab _·Ab_ invAb squash assocAb ridAb rinvAb commAb
+
+  -- The proof that incAb can be seen as a group homomorphism
+  incAbAsGroupHom : GroupHom G (AbGroup→Group asAbelianGroup)
+  incAbAsGroupHom = f , fIsHom
+    where
+    f = λ x → incAb x
+    fIsHom : IsGroupHom (snd G) f (snd (AbGroup→Group asAbelianGroup))
+    IsGroupHom.pres· fIsHom = λ x y → refl
+    IsGroupHom.pres1 fIsHom = refl
+    IsGroupHom.presinv fIsHom = λ x → refl
+
+  {- The proof of the universal property of the abelianization.
```

Comment:
The universal property could be in a separate module.

Reply:
done